src/test/java/org/apache/commons/lang3/StringUtilsChompTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.lang3;
+
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+import static org.apache.commons.lang3.StringUtils.length;
+
+/**
+ * Test to show whether using BitSet for removeAll() methods is faster than using HashSet.
+ */
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Thread)
+public class StringUtilsChompTest {
+
+    private static final String string1 = "\r";
+    private static final String string2 = buildString2();
+
+    private static String buildString2() {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < 100000; i++) {
+            stringBuilder.append('0');
+            stringBuilder.append('\r');
+            stringBuilder.append('0');
+            stringBuilder.append('\n');
+            stringBuilder.append('0');
+            stringBuilder.append('\r');
+            stringBuilder.append('\n');
+        }
+        return stringBuilder.toString();
+    }
+
+    @Benchmark
+    public void test1Old() {
+        chompOld(string1);
```

Comment:
You should return String from all your benchmark methods. Otherwise there is nothing to prevent the JVM from not running your chomp methods as the result is discarded. If you return the chomped String then the JMH framework will consume it thus forcing the JVM to run the method.

Reply:
@aherbert 
> Otherwise there is nothing to prevent the JVM from not running your chomp methods as the result is discarded. 
Is there some material/document about this?
I ask because seems I never observed this kind of thing happened...

Reply:
The JVM is free to discard code that does nothing useful.
https://stackoverflow.com/questions/4871550/how-can-i-be-sure-that-the-compiler-doesnt-optimize-away-my-performance-test

Reply:
JMH samples talk about this, e.g.: https://hg.openjdk.java.net/code-tools/jmh/file/4a420afe8a9e/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_08_DeadCode.java
https://hg.openjdk.java.net/code-tools/jmh/file/tip/jmh-samples/src/main/java/org/openjdk/jmh/samples/

Reply:
@sebbASF @thc202 thanks. remaked the jmh test.